### PR TITLE
fix: Fix bug with no SqlRegistryConfig class

### LINKED ISSUE #3544: SqlRegistryConfig does not exist
## Expected Behavior 
When specifying "sql" as the registry type via the python SDK, feast uses the appropriate config class.
## Current Behavior
When specifying "sql" as the registry type via the python SDK, calling the FeatureStore object results in this error:
`feast.errors.FeastClassImportError: Could not import class 'SqlRegistryConfig' from module 'feast.infra.registry.sql'` 

`get_registry_config_from_type` builds the class name here: https://github.com/feast-dev/feast/blob/master/sdk/python/feast/repo_config.py#L499 causing the config to look for the `SqlRegistryConfig` class, which doesn't exist. This is due to the code adding `Config` to the end of the current registry class, which works when specifying "file" (which uses the `RegistryConfig` class, which exists), but not for "sql".

## Steps to reproduce
Set my_repo_config = RepoConfig(registry=RepoConfig(registry_type="sql")) and then call FeatureStore(config=my_repo_config), which will then result in the error described above.

### Specifications

- Version: 0.30.0
- Platform: Ubuntu
- Subsystem:

## Possible Solution
Option 1:
Change the `REGISTRY_CLASS_FOR_TYPE` dict so that "sql" maps to "feast.infra.registry.registry.Registry" instead of "feast.infra.registry.sql.SqlRegistry" - This may also have to be done for "snowflake.registry"

Option 2:
Create the `SqlRegistryConfig` class in `feast.infra.registry.sql`. I'm also not seeing a `SnowflakeRegistryConfig` class in the `feast.infra.registry.snowflake` module, so that may need to be created as well.

Option 3:
Alter the `get_registry_config_from_type` function so it builds out the class names differently based on they registry type
https://github.com/feast-dev/feast/blob/master/sdk/python/feast/repo_config.py#L499

### PR DESCRIPTION
**Pull request by davidschuler-8451**
**What this PR does / why we need it**:
Fixes bug with using Sql Registry as described in #3544 

**Which issue(s) this PR fixes**:

Fixes #3544

### CODE REVIEW

**Comment by sudohainguyen** [comment]
+1

**Comment by sudohainguyen** [comment]
hey to simplify codebase, I think `SqlRegistryConfig` should be placed in `sql.py` instead, and it should inherit `RegistryConfig`

**Comment by davidschuler-8451** [comment]
> hey @davidschuler-8451 , to simplify codebase, I think `SqlRegistryConfig` should be placed in `sql.py` instead, and it should inherit `RegistryConfig`
> 
> I'm also working on this one, you can see my commit [here](https://github.com/feast-dev/feast/commit/1bbcfb59b9b3a73040c27d81fee3329b5ac37209) but feel free to apply to yours

thank you for that feedback! I went ahead and made those changes. Doesn't matter to me if your branch or mine gets merged in, just want to get this fixed because it's blocking some progress for us. Thanks!

**Comment by sudohainguyen** [comment]
ah your commits are required `DCO` to be ready for review, https://github.com/feast-dev/feast/pull/3586/checks?check_run_id=12566589779

**Comment by davidschuler-8451** [comment]
> ah your commits are required `DCO` to be ready for review, https://github.com/feast-dev/feast/pull/3586/checks?check_run_id=12566589779

yep - think i just fixed it. sorry about that

**Comment by sudohainguyen** [comment]
💯

**Review by adchia — APPROVED** [review]
/lgtm

**Comment by davidschuler-8451** [comment]
Sorry about the linter issue - i have my local env setup now (took a lot of trial and error with an M1 mac) so I can run the linting steps locally.

Might be a couple of ways to solve this:
1. we could let the SqlRegistry accept either a RegistryConfig or a SqlRegistryConfig like this. Doesn't feel like the cleanest option but it makes the linter happy
`registry_config: Union[Optional[SqlRegistryConfig], Optional[RegistryConfig]],`
2. convert the registry_config to an instance of SqlRegistryConfig here? https://github.com/feast-dev/feast/blob/master/sdk/python/feast/infra/registry/registry.py#L175

There may be other ways to do this as well, so I am open to suggestions. thank you!

File: sdk/python/feast/infra/registry/sql.py [hunk 1]
```diff
@@ -178,10 +179,19 @@ class FeastMetadataKeys(Enum):
...
+    path: StrictStr = ""
+    """ str: Path to metadata store.
+    If registry_type is 'sql', then this is a database URL as expected by SQLAlchemy """
+
+
 class SqlRegistry(BaseRegistry):
     def __init__(
         self,
-        registry_config: Optional[RegistryConfig],
+        registry_config: Optional[SqlRegistryConfig],
```
Comment (sudohainguyen) [inline sdk/python/feast/infra/registry/sql.py:194]: hey @davidschuler-8451  I have another idea, let's make this one `Union[RegistryConfig, SqlRegistryConfig]`

**Comment by davidschuler-8451** [comment]
@sudohainguyen & @adchia - fixed the issue with the build pipeline and things are looking good. Please review whenever you get a chance. thank you!

**Comment by davidschuler-8451** [comment]
thanks for the review @adchia! This is my first time opening a PR here - anything else I need to do to get this merged and released? Happy to help however necessary!

**Comment by sudohainguyen** [comment]
`lint-python` action getting stuck somehow, that's why this is not automatically merged by the bot

**Comment by davidschuler-8451** [comment]
hi @achals @felixwang9817 @kevjumba is anyone able to restart the linter action step so this can get merged in? This bug is affecting multiple teams and blocking some progress for us. See #3590, #3544. thank you!

**Pull request merged**